xarray/core/variable.py

```diff
@@ -893,16 +893,27 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
         if dim is not None and axis is not None:
             raise ValueError("cannot supply both 'axis' and 'dim' arguments")
 
+        if 'cum' in func.__name__:
```

Comment:
I don't think so unless there is a clever way to do this that I'm missing.  It looks like I'd need to have something like a new _partial_reduce_method that is for cumsum, cumprod, prod, etc.  However, this will require additional code lines beyond the simplistic approach I've taken.  But this may be necessary to get this to work in general in a clean way.  For instance, I don't think the existing code works properly with dataset yet.